lib/passport-tumblr/strategy.js

```diff
@@ -63,9 +64,8 @@ util.inherits(Strategy, OAuthStrategy);
  * @api protected
  */
 Strategy.prototype.authenticate = function (req, options) {
-  // Tumblr doesn't callback with any error parameter when users decline authorization
-  // but there is an `auth_token` after a success. (We first check the session, to see if we're after a callback.)
-  if (req.session[this._key] && !(req.query && req.query.auth_token)) return this.fail();
+  // Tumblr redirects to the callbackURL without any (error) parameter when users decline authorization
+  if (req.url === url.parse(this._callbackURL).path) return this.fail();
   OAuthStrategy.prototype.call(this, req, options);
```

Comment:
Should be

`OAuthStrategy.prototype.authenticate.call(this, req, options);`

??